modules/services/gpg-agent.nix

```diff
@@ -29,10 +29,44 @@ in
         '';
       };
 
+      defaultCacheTtlSsh = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Set the time a cache entry used for SSH keys is valid to the given number of seconds.
+        '';
+      };
+
       enableSshSupport = mkOption {
         type = types.bool;
         default = false;
-        description = "Whether to use the GnuPG key agent for SSH keys.";
+        description = ''
+          Whether to use the GnuPG key agent for SSH keys.
+        '';
+      };
+
+      noGrab = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Tell the pinentry not to grab the keyboard and mouse. This option should in general not be used to avoid X-sniffing attacks.
+        '';
+      };
+
+      disableScDaemon = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Do not make use of the scdaemon tool. This option has the effect of disabling the ability to do smartcard operations.
+        '';
+      };
+
+      writeEnvFile = mkOption {
```

Comment:
Are you sure this is needed? From what I can tell this option is no longer available:
```
WARNING: "--write-env-file" is an obsolete option - it has no effect
```

Reply:
I made these changes a while ago, could be it got deprecated.